aws_lambda_powertools/utilities/data_classes/code_pipeline_job_event.py

```diff
@@ -255,10 +274,53 @@ def get_artifact(self, artifact_name: str, filename: str) -> str | None:
         if artifact is None:
             return None
 
-        with tempfile.NamedTemporaryFile() as tmp_file:
-            s3 = self.setup_s3_client()
-            bucket = artifact.location.s3_location.bucket_name
-            key = artifact.location.s3_location.key
-            s3.download_file(bucket, key, tmp_file.name)
-            with zipfile.ZipFile(tmp_file.name, "r") as zip_file:
-                return zip_file.read(filename).decode("UTF-8")
+        s3 = self.setup_s3_client()
+        bucket = artifact.location.s3_location.bucket_name
+        key = artifact.location.s3_location.key
+
+        if filename:
+            with tempfile.NamedTemporaryFile() as tmp_file:
+                s3.download_file(bucket, key, tmp_file.name)
+                with zipfile.ZipFile(tmp_file.name, "r") as zip_file:
+                    return zip_file.read(filename).decode("UTF-8")
+
+        return s3.get_object(Bucket=bucket, Key=key)["Body"].read()
```

Comment:
I have added an option to get non-zip artifacts.

Reply:
Hey @mw-root! This looks like a super nice improvement to our codebase. 

I'm just curious if there is any situation where artifact is not a zip file, from my understanding, and of course I might be wrong, it's always a zip file. I'll try to find some documentation about this or even simulate this situation.

But anyway, this code looks great to me.

Reply:
I think as a general case AWS provided jobs will provide zip artifacts, but it's not a requirement. 
For Lambda jobs or custom stages you can upload whatever you want. The CodePipeline framework only defines the s3 location and provides credentials.

In one of our stages we call a Lambda and it uploads both zip and json artifacts that are used by later stages.

Reply:
Hey @mw-root! Yeah, thanks for explaining this, make sense! 

Resolving the conversation.